packages/core/src/services/time-series.ts

```diff
@@ -0,0 +1,140 @@
+];
+
+const codes = Object.values(locales).map(locale => locale["code"]);
+
+// Return true if the tick is a primary tick, false otherwise
+export function isTickPrimary(tick: number, i: number, interval: string, showDayName: boolean): boolean {
+	const isFirstTick = i === 0;
+	const hasANewWeekStarted = Number(format((new Date(tick)), "c")) === 2;
+	const isFirstQuarter = Number(format((new Date(tick)), "q")) === 1;
+
+	switch (interval) {
+		case "15seconds":
+			return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "minute":
+			return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "30minutes":
+			return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "hourly":
+			return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "daily":
+			if (!showDayName) { // daily
+				return (isFirstTick || isMonthChanged(tick) || isYearChanged(tick));
+			} else { // weekly
+				return (isFirstTick || hasANewWeekStarted || isYearChanged(tick));
+			}
+		case "monthly":
+			return (isFirstTick || isYearChanged(tick));
+		case "quarterly":
+			return (isFirstTick || isFirstQuarter);
+		case "yearly":
+			return false;
+		default:
+			throw new Error(`${interval} is not a valid time interval.`);
+	}
+}
+
+// The accepted formats of localeCode are ll and ll-CC
+function getLocale(localeCode: string): Locale {
+	// locales is an object whose keys format is ll or llCC
+	// each locale is an object whose code value format is ll or ll-CC
+	const localeCodeWithoutDash = localeCode.replace(/-/g, "");
+	const foundLocale = locales[localeCodeWithoutDash];
+	if (!foundLocale) {
+		throw new Error(`Locale with code ${localeCode} not found. Avaible codes are: ${codes}.`);
+	}
+	return foundLocale;
+}
+
+// Return the formatted current tick
+export function formatTick(tick: number, i: number, interval: string, timeScaleOptions: TimeScaleOptions): string {
+	const showDayName = timeScaleOptions.showDayName;
+	const intervalConsideringAlsoShowDayNameOption = interval === "daily" && showDayName ? "weekly" : interval;
+	const date = new Date(tick);
+	const formats = Tools.getProperty(timeScaleOptions, "timeIntervalFormats")[intervalConsideringAlsoShowDayNameOption];
+	const primary = Tools.getProperty(formats, "primary");
+	const secondary = Tools.getProperty(formats, "secondary");
+	const localeCode = timeScaleOptions.localeCode;
+	const formatString = isTickPrimary(tick, i, interval, showDayName) ? primary : secondary;
+	const locale = getLocale(localeCode);
+
+	return format(date, formatString, { locale });
+}
+
+// Given a timestamp, return an object of useful time formats
+// Use Unicode date field symbol (https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table)
+export function timestampToFormatTime(timestamp: number) {
```

Comment:
Can we think of a more readable name here?
I like your description in the comments above.

Maybe `getTimeformats(timestamp: number)`?

Reply:
Sure, you're right.